3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -22,18 +22,96 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+const URL = 'https://pokeapi.co/api/v2/pokemon/';
+
+function createGetButton() {
+  const button = document.createElement('button');
+  button.type = 'button';
+  button.className = 'button';
+  button.textContent = 'Get Pokemon!';
+
+  return button;
+}
+
+function createSelectElement() {
+  const selectList = document.createElement('select');
+  selectList.name = 'Pokemons';
+  selectList.id = 'select';
+
+  return selectList;
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function createUserInterface() {
+  const container = document.createElement('div');
+  container.id = 'user-interface';
+  const sidebar = document.createElement('div');
+  sidebar.className = 'sidebar';
+  const button = createGetButton();
+  const selectList = createSelectElement();
+  const imgContainer = document.createElement('div');
+  const img = document.createElement('img');
+  imgContainer.className = 'image-container';
+  imgContainer.appendChild(img);
+  sidebar.appendChild(button);
+  sidebar.appendChild(selectList);
+  container.appendChild(sidebar);
+  container.appendChild(imgContainer);
+  document.body.appendChild(container);
 }
 
-function fetchImage(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchData(url) {
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error('Request failed!', response.status, response.statusText);
+  }
+  const jsonResponse = await response.json();
+  return jsonResponse;
 }
 
-function main() {
-  // TODO complete this function
+async function fetchAndPopulatePokemons(pokemons) {
+  const selectList = document.querySelector('#select');
+  const getButton = document.querySelector('button');
+  const renderSelectList = () => {
+    pokemons.forEach((pokemon) => {
+      const option = document.createElement('option');
+      option.value = pokemon.name;
```

Comment:
```suggestion
      option.value = pokemon.url;
```

Each pokemon object has a `url` property that can be used to fetch the image. This is the url you should actually use. By assigning it to the `option.value` attribute you can pick it up later to fetch the image in the `change` event handler.